lib/rules/declaration-block-no-redundant-longhand-properties/index.js

```diff
@@ -96,6 +96,12 @@ const customResolvers = new Map([
 
 			if (!(areas && columns && rows)) return;
 
+			// repeat() is not allowed inside track listings for grid-template.
+			// related issue: https://github.com/stylelint/stylelint/issues/7228
+			// spec ref: https://drafts.csswg.org/css-grid/#explicit-grid-shorthand
+
+			if (columns.includes('repeat') || rows.includes('repeat')) return;
```

Comment:
[suggest] For more accuracy, how about including "("?

```suggestion
			if (columns.includes('repeat(') || rows.includes('repeat(')) return;
```